data-raw/build_gene_reference.Rmd

```diff
@@ -0,0 +1,179 @@
+---
+title: "Gene ID Conversion data setup"
+output: html_notebook
+---
+
+
+This notebook is to set up the reference data for gene ID conversion and ultimately deduplication.
+
+```{r}
+suppressPackageStartupMessages({
+  library(SingleCellExperiment)
+  library(dplyr)
+})
+```
+
+
+Read in the test data ScPCA SCE object and extract the row data:
+
+```{r}
+genes_scpca <- readRDS(here::here("tests", "testthat", "data", "scpca_sce.rds")) |>
+  rowData() |>
+  as.data.frame() |>
+  # Use Ensembl ID if gene symbol is missing, then make unique
+  mutate(
+    gene_symbol_scpca = ifelse(is.na(gene_symbol), gene_ids, gene_symbol),
+    gene_symbol_scpca_unique = make.unique(gene_symbol_scpca)
+  ) |>
+  select(gene_ids, gene_symbol_scpca, gene_symbol_scpca_unique)
+```
+
+
+Download and read in a 2020 10x reference dataset and extract the gene symbols.
+Note that the 2020 Cell Ranger reference does not use Ensembl gene IDs for missing symbols, but the 2024 reference does.
+
+```{r}
+# Download data
+url_10x2020 <- "https://cf.10xgenomics.com/samples/cell-exp/7.0.1/SC3pv3_GEX_Human_PBMC/SC3pv3_GEX_Human_PBMC_filtered_feature_bc_matrix.h5" # nolint
+temp_10x2020 <- tempfile(fileext = ".h5")
+download.file(url_10x2020, temp_10x2020, mode = "wb")
+on.exit(unlink(temp_10x2020), add = TRUE) # delete when done
+```
+
+```{r}
+# Read in the data
+genes_10x2020 <- DropletUtils::read10xCounts(temp_10x2020) |>
+  rowData() |>
+  as.data.frame() |>
+  filter(Type == "Gene Expression") |>
+  rename(
+    gene_ids = ID,
+    gene_symbol_10x2020 = Symbol
+  ) |>
+  # Use Ensembl ID if gene symbol is missing, then make unique
+  mutate(gene_symbol_10x2020_unique = make.unique(gene_symbol_10x2020)) |>
+  select(gene_ids, gene_symbol_10x2020, gene_symbol_10x2020_unique)
+```
+Download and read in a 2024 10x reference dataset and extract the gene symbols:
+
+```{r}
+# Download data
+url_10x2024 <- "https://cf.10xgenomics.com/samples/cell-exp/9.0.0/5k_Human_Donor2_PBMC_3p_gem-x_5k_Human_Donor2_PBMC_3p_gem-x/5k_Human_Donor2_PBMC_3p_gem-x_5k_Human_Donor2_PBMC_3p_gem-x_count_sample_filtered_feature_bc_matrix.h5" # nolint
+temp_10x2024 <- tempfile(fileext = ".h5")
+download.file(url_10x2024, temp_10x2024, mode = "wb")
+on.exit(unlink(temp_10x2024), add = TRUE) # delete when done
+```
+
+
+```{r}
+genes_10x2024 <- DropletUtils::read10xCounts(temp_10x2024) |>
+  rowData() |>
+  as.data.frame() |>
+  filter(Type == "Gene Expression") |>
+  rename(
+    gene_ids = ID,
+    gene_symbol_10x2024 = Symbol
+  ) |>
+  # Use Ensembl ID if gene symbol is missing, then make unique
+  mutate(gene_symbol_10x2024_unique = make.unique(gene_symbol_10x2024)) |>
+  select(gene_ids, gene_symbol_10x2024, gene_symbol_10x2024_unique)
+```
+
+## Merge the gene symbol tables
+
+```{r}
+scpca_gene_reference <- genes_scpca |>
+  full_join(genes_10x2020, by = "gene_ids") |>
+  full_join(genes_10x2024, by = "gene_ids")
+```
+
+
+## Look at some stats for the tables
```

Comment:
At some point later" I got a little bit lost in keeping in my brain "are we checking for scpca ids not in 10x or vice versa?" But, I think you've written everything clearly! Just my brain keeping track, albeit late in a long day.... 

Either way, maybe a slightly more informative header here (and/or subheaders below?) might help future brains stay on top of what's going on.